packages/block-library/src/post-featured-image/index.php

```diff
@@ -89,6 +89,17 @@ function render_block_core_post_featured_image( $attributes, $content, $block )
 			foreach ( $processor->get_attribute_names_with_prefix( '' ) as $name ) {
 				$tag_html->set_attribute( $name, $processor->get_attribute( $name ) );
 			}
+
+			foreach ( $attr as $name => $value ) {
+				if ( 'style' !== $name ) {
+					$tag_html->set_attribute( $name, $value );
+				} else {
+					$existing_styles = $tag_html->get_attribute( 'style' );
+					$new_styles      = empty( $existing_styles ) ? $value : $existing_styles . ';' . $value;
+					$tag_html->set_attribute( 'style', $new_styles );
+				}
+			}
```

Comment:
@t-hamano a call to `set_attribute()` fully replaces the existing value, so there's no need to call `remove_attribute()` first.

@rohitmathur-7 there can be strange interactions with inline styles. a new style at the end is not guaranteed to merge in a specific way with the previously-set values. this is mostly evident when appending properties like `border` where existing border attributes are already set. this is because some properties incorporate multiple properties in shorthand notation, e.g. `border: 1px solid blue; border-left: red;`. just a thing to consider here.